domain/userDB.go

```diff
@@ -0,0 +1,17 @@
+package domain 
+
+//UserDB represents a user database with file pointers and segments based on search results
+type UserDB struct {
+	//Document DB pointer
+    FilePointers [] string
+
+	//Vector search segments
+	Segments [] string
```

Comment:
`[] string` is not valid go syntax for go [slices](https://go.dev/tour/moretypes/7)

Consider changing to 
```suggestion
	Segments []string
```

Reply:
Fixed good call. I think adding GOFMT to pre-hooks will help negate some of this later down the line. I should probably get a ticket board for these typa thing. Will follow up